cpp-package/example/mlp_csv.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Amol Lele amol.github@gmail.com
+ */
+
+/*
+ * Example: mlp_csv_cpu
+ * Description:
+ * The following example demonstrates how to use CSVIter. This example creates
+ * mlp (multi-layer perceptron) model and trains the MNIST data which is in
+ * CSV format.
+ */
+#include <chrono>
+#include "utils.h"
+#include "mxnet-cpp/MxNetCpp.h"
+
+using namespace mxnet::cpp;
+
+/*
+ * Implementing the mlp symbol with given hidden units configuration.
+ */
+Symbol mlp(const std::vector<int> &hidden_units)
+{
+    auto data = Symbol::Variable("data");
+    auto label = Symbol::Variable("label");
+
+    std::vector<Symbol> weights(hidden_units.size());
+    std::vector<Symbol> biases(hidden_units.size());
+    std::vector<Symbol> outputs(hidden_units.size());
+
+    for (size_t i = 0; i < hidden_units.size(); ++i) {
+        weights[i] = Symbol::Variable("w" + std::to_string(i));
+        biases[i] = Symbol::Variable("b" + std::to_string(i));
+        Symbol fc = FullyConnected(
+                                   i == 0? data : outputs[i-1],  // data
+                                   weights[i],
+                                   biases[i],
+                                   hidden_units[i]);
+        outputs[i] = i == hidden_units.size()-1 ? fc : Activation(fc, ActivationActType::kRelu);
+    }
+    return SoftmaxOutput(outputs.back(), label);
+}
+
+/*
+ * Convert the input string of number of hidden units into the vector of integers.
+ */
+void getLayers(std::string &hidden_units_string, std::vector<int> &hidden_units)
+{
+    std::string delimiter = ",";
+    size_t pos = 0;
+    std::string token;
+    while ((pos = hidden_units_string.find(delimiter)) != std::string::npos) {
+        token = hidden_units_string.substr(0, pos);
+        hidden_units.push_back(atoi(token.c_str()));
+        hidden_units_string.erase(0, pos + delimiter.length());
+    }
+    hidden_units.push_back(atoi(hidden_units_string.c_str()));
+}
+
+void printUsage()
```

Comment:
do you want to pull out the arg parsing to a different file to keep this clean? may be you can generalize arg parsing later as you add new examples to C++

Reply:
I would defer pulling out or refactoring it.  As we make more changes to these examples, we will get better idea and we can come up with more generic command line parser.